### PR TITLE
DEV-5216 | Make BadActorSerializer more idiomatic

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### What's this PR do?

This PR refactors `BadActorSerializer`, removing an over-ride of `to_internal_value`. As noted in ticket for this PR, I discovered the following oddity:

```
from apps.contributions.serializers import BadActorSerializer

instance  = BadActorSerializer(data={"first_name": "foo", "last_name": "bar", "email": "foo@bar.com", "amount": "12400", "ip":  "192.0.2.1", "referer": "https://fundjournalism.org" })
instance.is_valid()
# false
instance.errors
Out[4]: {'street': [ErrorDetail(string='This field may not be null.', code='null')], '.....
```

Not totally clear why, but somehow using the over-ride of to_internval_value cased the default values for those fields to not be present, triggering a validation error.

Unclear if any bugs directly resulted from that, but it causes the serializer to behave in unexpected ways.

To solve for this, instead of mapping source values to destination values in `to_internal_value` we just create input data with necessary keys ahead of initializing serializer.

#### Why are we doing this? How does it help us?

Make BadActorSerializer's behavior more predictable

#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?

Yes

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No

#### Have automated unit tests been added? If not, why?

No. We remove existing code.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-5216](https://news-revenue-hub.atlassian.net/browse/DEV-5216)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:

No

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No

[DEV-5216]: https://news-revenue-hub.atlassian.net/browse/DEV-5216?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/contributions/serializers.py [hunk 1]
```diff
@@ -154,32 +154,29 @@ class BadActorSerializer(serializers.Serializer):
     org = serializers.CharField(max_length=255, required=False, default="", allow_blank=True)
     street = serializers.CharField(max_length=255, required=False, default="", allow_blank=True)
     complement = serializers.CharField(max_length=255, required=False, default="", allow_blank=True)
-    city = serializers.CharField(max_length=40, required=False, default="", allow_blank=True)
-    state = serializers.CharField(max_length=80, required=False, default="", allow_blank=True)
+    city = serializers.CharField(
```
Comment (njharman) [inline apps/contributions/serializers.py:157]: I don't see any changes in these lines. Wonder why formatter chose to break apart, ... Oh, probably the trailing comma.

I find that annoying.

File: apps/contributions/serializers.py [hunk 2]
```diff
@@ -399,23 +396,29 @@ def get_bad_actor_score(self, data, action: BadActorAction) -> BadActorOverallSc
 
         Note that this method is meant to be ironclad against exceptions. If anything goes wrong, it should return None.
         """
+        data = {
```
Comment (njharman) [inline apps/contributions/serializers.py:399]: Given docstring 
> Note that this method is meant to be ironclad against exceptions. If anything goes wrong, it should return None.

All of this should be in try block

Comment (njharman) [inline apps/contributions/serializers.py:399]: There's a typo in the catch Exception "sore" should be "score".  But real reason for comment is consider adding debugging info to exception log. Such as "data" (before it's modified)

File: apps/contributions/serializers.py [hunk 3]
```diff
@@ -399,23 +396,29 @@ def get_bad_actor_score(self, data, action: BadActorAction) -> BadActorOverallSc
...
+        }
+        if country_code := self.context["request"].headers.get("Cf-Ipcountry", None):
+            data["country_code"] = country_code
+        logger.info("BadActorSerializer data: %s", data)
+        serializer = BadActorSerializer(data=data)
         try:
-            data = data | {
-                # we use a PrimaryKeyRelated serializer field for page in BaseCreatePaymentSerializer
-                # but BadActorSerializer wants to pk, so we reformat here.
-                "page": data["page"].id,
```
Comment (njharman) [inline apps/contributions/serializers.py:406]: I don't see "page" replicated above.

Was it intentional removal?

File: apps/contributions/serializers.py (hunk 2 above)
Comment (njharman) [inline apps/contributions/serializers.py:399]: Suggest not reusing name "data" also used as argument.

**Review by njharman — CHANGES_REQUESTED** [review]

File: apps/contributions/serializers.py [hunk 4]
```diff
@@ -399,23 +396,29 @@ def get_bad_actor_score(self, data, action: BadActorAction) -> BadActorOverallSc
...
         """
+        data = {
+            "amount": data["amount"],
+            "first_name": data["first_name"],
+            "last_name": data["last_name"],
+            "email": data["email"],
+            "action": action,
+            "org": data["page"].organization.id,
+            "street": data["mailing_street"],
+            "complement": data["mailing_complement"],
```
Comment (njharman) [inline apps/contributions/serializers.py:407]: complement, captcha_token, ip, referer, reason_for_giving don't seem to be tested for in `test_get_bad_actor_score_fields()`

But there probably should be a separate test to prove AC2. Something with all default values and none of those defaults are `None`. At least that is my understanding of issue

> string='This field may not be null.'

**Review by njharman — CHANGES_REQUESTED** [review]

File: apps/contributions/serializers.py (hunk 3 above)
Comment (benjaminEwhite) [reply apps/contributions/serializers.py:406]: And that's why we have code review...

File: apps/contributions/serializers.py (hunk 2 above)
Comment (benjaminEwhite) [reply apps/contributions/serializers.py]: 25c2a0e899

File: apps/contributions/serializers.py (hunk 4 above)
Comment (benjaminEwhite) [reply apps/contributions/serializers.py]: fd41aec9ff0

File: apps/contributions/serializers.py (hunk 2 above)
Comment (benjaminEwhite) [reply apps/contributions/serializers.py]: 1abbbc571

Comment (benjaminEwhite) [reply apps/contributions/serializers.py]: Resolved the typo, but not sure which commit.

RE: adding data d59fb117e49b

Comment (benjaminEwhite) [reply apps/contributions/serializers.py]: Resolved the typo, but not sure in which commit.

Re: data and logging d59fb117e49b

**Review by nrh-cklimas — APPROVED** [review]

**Pull request merged**